### ISSUE TITLE
训练出来的视频有点抖动

### ISSUE THREAD

**Issue by wanghx1121** [ZiqiaoPeng/SyncTalk#70]
训练出来的视频有点抖动，麻烦看一下，是否可以优化。谢谢！~

https://github.com/ZiqiaoPeng/SyncTalk/assets/159092899/1a6b6b5b-5cae-4629-b19f-8a48e8b24175

@ZiqiaoPeng

**Comment by ZiqiaoPeng** [comment]
重新pull一下代码再试一下，刚修复了几个bug。

**Comment by wanghx1121** [comment]
> 重新pull一下代码再试一下，刚修复了几个bug。重新运行一下数据预处理的过程。

好的，谢谢

**Comment by wanghx1121** [comment]
> > 训练出来的视频有点抖动，麻烦看一下，是否可以优化。谢谢！~
> > ngp_ep0063_audio.mp4
> > @ZiqiaoPeng
> 
> 你好，请问下你的音频选用的什么模型，最终loss大概多少收敛呢

loss = 0.0052,使用以下命令训练的：
python main.py data/m2fp --workspace model/trial_m2fp -O --iters 60000 --asr_model ave
python main.py data/m2fp --workspace model/trial_m2fp -O --iters 100000 --finetune_lips --patch_size 64 --asr_model ave

**Comment by wanghx1121** [comment]
https://github.com/ZiqiaoPeng/SyncTalk/assets/159092899/6614dc83-63a1-47a3-b7f7-83fe165605be

@ZiqiaoPeng  你好，现在效果好很多。但是，脖子处好像有断层。是否可以优化一下。谢谢！~

**Comment by ZiqiaoPeng** [comment]
这是正常的，第二阶段加入了额外的loss约束，占比较大，所以没有出现明显的下降趋势。最终结果看评价指标会更合理一些。

**Comment by wennjiee** [comment]
> 这是正常的，第二阶段加入了额外的loss约束，占比较大，所以没有出现明显的下降趋势。最终结果看评价指标会更合理一些。

感谢！

**Comment by wanghx1121** [comment]
> 这是正常的，第二阶段加入了额外的loss约束，占比较大，所以没有出现明显的下降趋势。最终结果看评价指标会更合理一些。

@ZiqiaoPeng  你好，我看了一下。所有步数生成的模型都会保留。程序最终会选择最优模型吗？

**Comment by wanghx1121** [comment]
> > > > 训练出来的视频有点抖动，麻烦看一下，是否可以优化。谢谢！~
> > > > ngp_ep0063_audio.mp4
> > > > @ZiqiaoPeng
> > > 
> > > 
> > > 你好，请问下你的音频选用的什么模型，最终loss大概多少收敛呢
> > 
> > 
> > loss = 0.0052,使用以下命令训练的： python main.py data/m2fp --workspace model/trial_m2fp -O --iters 60000 --asr_model ave python main.py data/m2fp --workspace model/trial_m2fp -O --iters 100000 --finetune_lips --patch_size 64 --asr_model ave
> 
> 你好，可以看下图，我第一阶段60k步之前loss和你差不多，第二阶段60k步后微调时loss显著上升，似乎收敛在0.02，我目前不太清楚这是否是正常现象，正在学习这部分代码。 ![image](https://private-user-images.githubusercontent.com/68308198/326438085-a52c9143-7682-4f23-97ee-66cb744c382e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ0Mzc1MjksIm5iZiI6MTcxNDQzNzIyOSwicGF0aCI6Ii82ODMwODE5OC8zMjY0MzgwODUtYTUyYzkxNDMtNzY4Mi00ZjIzLTk3ZWUtNjZjYjc0NGMzODJlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDMwVDAwMzM0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY5NGFhMjhmNmViYTRhOWEwNTg3MjRmZWU3ZDNmOTgxNDYxZTQ2OGIzMzFlOGM5MmY0Y2ExM2YzNzhhOWUyNDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ZTrMhlWpicnRi9ExZTYOAuxtMBB1apUhAScfp1uQ-6w)

@wennjiee 这个图表，怎么生成的。？

**Comment by wennjiee** [comment]
> > > > > 训练出来的视频有点抖动，麻烦看一下，是否可以优化。谢谢！~
> > > > > ngp_ep0063_audio.mp4
> > > > > @ZiqiaoPeng
> > > > 
> > > > 
> > > > 你好，请问下你的音频选用的什么模型，最终loss大概多少收敛呢
> > > 
> > > 
> > > loss = 0.0052,使用以下命令训练的： python main.py data/m2fp --workspace model/trial_m2fp -O --iters 60000 --asr_model ave python main.py data/m2fp --workspace model/trial_m2fp -O --iters 100000 --finetune_lips --patch_size 64 --asr_model ave
> > 
> > 
> > 你好，可以看下图，我第一阶段60k步之前loss和你差不多，第二阶段60k步后微调时loss显著上升，似乎收敛在0.02，我目前不太清楚这是否是正常现象，正在学习这部分代码。 ![image](https://private-user-images.githubusercontent.com/68308198/326438085-a52c9143-7682-4f23-97ee-66cb744c382e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ0Mzc1MjksIm5iZiI6MTcxNDQzNzIyOSwicGF0aCI6Ii82ODMwODE5OC8zMjY0MzgwODUtYTUyYzkxNDMtNzY4Mi00ZjIzLTk3ZWUtNjZjYjc0NGMzODJlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDMwVDAwMzM0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY5NGFhMjhmNmViYTRhOWEwNTg3MjRmZWU3ZDNmOTgxNDYxZTQ2OGIzMzFlOGM5MmY0Y2ExM2YzNzhhOWUyNDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ZTrMhlWpicnRi9ExZTYOAuxtMBB1apUhAScfp1uQ-6w)
> 
> @wennjiee 这个图表，怎么生成的。？

使用的tensorboard进行查看，可以参照相关使用方法

**Comment by wanghx1121** [comment]
> > ngp_ep0063_audio.mp4
> > @ZiqiaoPeng Hello, the effect is much better now. However, there seems to be a break in the neck. Can it be optimized? Thanks! ~
> 
> May ask the iterations amount you used for this example? And dataset size or video length? From my test, 60000 + 40000 was not enough for 5k frames dataset. I am now at 200k and 300k, should be done soon.

hi,I trained using the default number of steps provided by the author.

**Comment by wanghx1121** [comment]
> > > > > > 训练出来的视频有点抖动，麻烦看一下，是否可以优化。谢谢！~
> > > > > > ngp_ep0063_audio.mp4
> > > > > > @ZiqiaoPeng
> > > > > 
> > > > > 
> > > > > 你好，请问下你的音频选用的什么模型，最终loss大概多少收敛呢
> > > > 
> > > > 
> > > > loss = 0.0052,使用以下命令训练的： python main.py data/m2fp --workspace model/trial_m2fp -O --iters 60000 --asr_model ave python main.py data/m2fp --workspace model/trial_m2fp -O --iters 100000 --finetune_lips --patch_size 64 --asr_model ave
> > > 
> > > 
> > > 你好，可以看下图，我第一阶段60k步之前loss和你差不多，第二阶段60k步后微调时loss显著上升，似乎收敛在0.02，我目前不太清楚这是否是正常现象，正在学习这部分代码。 ![image](https://private-user-images.githubusercontent.com/68308198/326438085-a52c9143-7682-4f23-97ee-66cb744c382e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ0Mzc1MjksIm5iZiI6MTcxNDQzNzIyOSwicGF0aCI6Ii82ODMwODE5OC8zMjY0MzgwODUtYTUyYzkxNDMtNzY4Mi00ZjIzLTk3ZWUtNjZjYjc0NGMzODJlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDMwVDAwMzM0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY5NGFhMjhmNmViYTRhOWEwNTg3MjRmZWU3ZDNmOTgxNDYxZTQ2OGIzMzFlOGM5MmY0Y2ExM2YzNzhhOWUyNDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ZTrMhlWpicnRi9ExZTYOAuxtMBB1apUhAScfp1uQ-6w)
> > 
> > 
> > @wennjiee 这个图表，怎么生成的。？
> 
> 使用的tensorboard进行查看，可以参照相关使用方法

@wennjiee 日志文件指定log_ngp.txt可以吗？

**Comment by wanghx1121** [comment]
> ngp_ep0063_audio.mp4 
> @ZiqiaoPeng 你好，现在效果好很多。但是，脖子处好像有断层。是否可以优化一下。谢谢！~

![image](https://github.com/ZiqiaoPeng/SyncTalk/assets/159092899/9172f9ee-720e-4e76-a82c-2d38fbe67736)

python main.py data/m2fp --workspace model/trial_m2fp -O --iters 60000 --asr_model ave python main.py data/m2fp --workspace model/trial_m2fp -O --iters 100000 --finetune_lips --patch_size 64 --asr_model ave

@ZiqiaoPeng @wennjiee  两位帮忙看一下，这是我使用上述命令训练生成的梯度图，是否需要加大步数？

**Comment by yulj21** [comment]
> ngp_ep0063_audio.mp4 
> @ZiqiaoPeng 你好，现在效果好很多。但是，脖子处好像有断层。是否可以优化一下。谢谢！~

您好， 请问这个视频训练样本的帧数有多少，训练epoch是63个吗

**Comment by faiejir** [comment]
> 这是正常的，第二阶段加入了额外的loss约束，占比较大，所以没有出现明显的下降趋势。最终结果看评价指标会更合理一些。

请问如何看评价指标，应该如何参照
